phys/module_diag_afwa.F

```diff
@@ -236,28 +311,32 @@ SUBROUTINE afwa_diagnostics_driver (   grid , config_flags     &
 
     ! Calculate precipitable water
     ! ----------------------------
-    nz=kme-kms+1
-    DO i=ims,ime
-      DO j=jms,jme
-        grid % afwa_pwat ( i, j ) = Pwat( nz,                  &
-                                          qvapr(i,kms:kme,j),  &
-                                          qcloud(i,kms:kme,j), &
-                                          dz8w(i,kms:kme,j),   &
-                                          rho(i,kms:kme,j) )
+    IF ( F_QV .AND. F_QC ) THEN
+      nz=kme-kms+1
+      DO i=ims,ime
+        DO j=jms,jme
+          grid % afwa_pwat ( i, j ) = Pwat( nz,                  &
+                                            qvapr(i,kms:kme,j),  &
+                                            qcloud(i,kms:kme,j), &
+                                            dz8w(i,kms:kme,j),   &
+                                            rho(i,kms:kme,j) )
+        ENDDO
       ENDDO
-    ENDDO
+    ELSE
+      CALL wrf_debug ( DEBUG_LEVEL, 'Option pwat requires: QV, QC' )
+    END IF
 
     ! After each history dump, reset max/min value arrays
     ! ----------------------------------------------------------------------
     IF ( is_after_history_dump ) THEN
-      DO j = jms, jme
-        DO i = ims, ime
-          grid % wspd10max(i,j) = 0.
-          IF ( config_flags % afwa_severe_opt == 1 ) THEN
-          grid % afwa_llws(i,j) = 0.
-          ENDIF
+      IF ( config_flags % afwa_severe_opt == 1 ) THEN
+        DO j = jms, jme
+          DO i = ims, ime
+            grid % wspd10max(i,j) = 0.
+            grid % afwa_llws(i,j) = 0.
+          ENDDO
         ENDDO
-      ENDDO
+      ENDIF
     ENDIF 
 
     ! Calculate the max 10 m wind speed between output times
```

Comment:
@davegill The do loop just after these comments need to be put in 
IF ( config_flags % afwa_severe_opt == 1 ) THEN
ENDIF

Reply:
@weiwangncar 
Wei,
Sorry, put the comment in the wrong place. I think I fixed this now by including an IF test around this larger block that encompasses the wspd10max location.